### ISSUE TITLE
Bellman loss function automated tests contain redundancies and confusing comments

### ISSUE THREAD

**Issue by sbenthall** [scikit-agent/scikit-agent#100]
The automated tests for the Bellman loss function in `test_maliar.py` are extensive but seem to contain redundancies and nonsensical comments. AI generated?

These three tests are identical, though the comments make it seem like they are doing different things:
https://github.com/scikit-agent/scikit-agent/blob/main/tests/test_maliar.py#L471-L551

This test says it's about 'integration with training loop components' but seems to mainly just test that the loss is different if a different decision rule is used:
https://github.com/scikit-agent/scikit-agent/blob/main/tests/test_maliar.py#L610C1-L652C54

This sort of bloat makes it harder to use the tests to understand how to use the software.
